### PR TITLE
Add validation tests for ESM2 fine-tuning benchmark partial-conv

### PR DESCRIPTION
**Pull request by nvmvle**
### Description
This PR adds a comprehensive test validation section to the ESM2 fine-tuning benchmark partial-conv configuration. The tests define expected baseline metrics and ranges for model performance validation during CI runs.

The validation includes:
- Exit code validation (expecting successful completion with code 0)
- Performance baselines for key metrics:
  - `consumed_samples`
  - `val_loss`
  - `val_acc`
  - `TFLOPS per GPU`
  - `reduced_train_loss`

### Type of changes
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [x]  New feature (non-breaking change which adds functionality)
- [ ]  Refactor
- [ ]  Documentation update
- [ ]  Other (please describe):

### CI Pipeline Configuration
Configure CI behavior by applying the relevant labels:

- [SKIP_CI](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#skip_ci) - Skip all continuous integration tests
- [INCLUDE_NOTEBOOKS_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_notebooks_tests) - Execute notebook validation tests in pytest
- [INCLUDE_SLOW_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_slow_tests) - Execute tests labelled as slow in pytest for extensive testing

> [!NOTE]
> By default, the notebooks validation tests are skipped unless explicitly enabled.

#### Authorizing CI Runs

We use [copy-pr-bot](https://docs.gha-runners.nvidia.com/apps/copy-pr-bot/#automation) to manage authorization of CI
runs on NVIDIA's compute resources.

* If a pull request is opened by a trusted user and contains only trusted changes, the pull request's code will
  automatically be copied to a pull-request/ prefixed branch in the source repository (e.g. pull-request/123)
* If a pull request is opened by an untrusted user or contains untrusted changes, an NVIDIA org member must leave an
  `/ok to test` comment on the pull request to trigger CI. This will need to be done for each new commit.

### Pre-submit Checklist

 - [x] I have tested these changes locally
 - [ ] I have updated the documentation accordingly
 - [ ] I have added/updated tests as needed
 - [x] All existing tests pass successfully
 
 Signed-off-by: My Le <mvle@nvidia.com>

### CODE REVIEW

**Review by pstjohn — APPROVED** [review]

**Review by jwilber — APPROVED** [review]

**Comment by nvmvle** [comment]
/ok to test https://github.com/NVIDIA/bionemo-framework/pull/1010/commits/dfd69136979db4e2a19082fcb90740d64b4aa00e

**Comment by nvmvle** [comment]
/ok to test https://github.com/NVIDIA/bionemo-framework/pull/1010/commits/42918f649237dc849048b6f5c993c5e281601a5b
